tests/components/dsmr/test_config_flow.py

```diff
@@ -1,12 +1,82 @@
 """Test the DSMR config flow."""
+import asyncio
+
+from dsmr_parser.clients.protocol import DSMRProtocol
+from dsmr_parser.obis_references import EQUIPMENT_IDENTIFIER, EQUIPMENT_IDENTIFIER_GAS
+from dsmr_parser.objects import CosemObject
+import pytest
+import serial
+
 from homeassistant import config_entries, setup
 from homeassistant.components.dsmr import DOMAIN
 
-from tests.async_mock import patch
+import tests.async_mock
+from tests.async_mock import Mock, patch
 from tests.common import MockConfigEntry
 
+SERIAL_DATA = {"serial_id": "12345678", "serial_id_gas": "123456789"}
+
+
+@pytest.fixture
+def mock_connection_factory(monkeypatch):
+    """Mock the create functions for serial and TCP Asyncio connections."""
+    transport = tests.async_mock.Mock(spec=asyncio.Transport)
+    protocol = tests.async_mock.Mock(spec=DSMRProtocol)
+
+    async def connection_factory(*args, **kwargs):
+        """Return mocked out Asyncio classes."""
+        return (transport, protocol)
+
+    connection_factory = Mock(wraps=connection_factory)
+
+    # apply the mock to both connection factories
+    monkeypatch.setattr(
```

Comment:
wonder if we should bd using patch.object here vs monkeypatch

Reply:
Our standard is to use `patch` imported from `tests.async_mock`.

Reply:
This mock function is copy-paste from `test_sensor.py`, which also deviates from standard. I think generally tests can be cleaned up a bit as they seem to use different ways then I have seen before. Would ik be ok to leave this in and address refactoring of tests in a follow-up PR?

Reply:
Yes, that's ok.